client/src/modules/stock/StockFilterer.service.js

```diff
@@ -0,0 +1,145 @@
+angular.module('bhima.services')
+  .service('StockFilterer', StockFiltererService);
+
+StockFiltererService.$inject = [
+  'FilterService', 'appcache', 'PeriodService',
+  '$httpParamSerializer', 'LanguageService', 'bhConstants', 'util',
+];
+
+/**
+ * @class StockFilterer
+ */
+function StockFiltererService(Filters, AppCache, Periods, $httpParamSerializer, Languages, bhConstants, Util) {
+  const customFiltersList = [
+    { key : 'depot_uuid', label : 'STOCK.DEPOT' },
+    { key : 'inventory_uuid', label : 'STOCK.INVENTORY' },
+    { key : 'group_uuid', label : 'STOCK.INVENTORY_GROUP' },
+    { key : 'label', label : 'STOCK.LOT' },
+    { key : 'is_exit', label : 'STOCK.OUTPUT' },
+    { key : 'reference', label : 'FORM.LABELS.REFERENCE' },
+    { key : 'flux_id', label : 'STOCK.FLUX' },
+    { key : 'user_id', label : 'FORM.LABELS.USER' },
+    { key : 'status', label : 'STOCK.STATUS.LABEL', valueFilter : 'translate' },
+    { key : 'require_po', label : 'STOCK.REQUIRES_PO' },
+    {
+      key : 'dateFrom', label : 'FORM.LABELS.DATE', comparitor : '>', valueFilter : 'date',
+    },
+    {
+      key : 'dateTo', label : 'FORM.LABELS.DATE', comparitor : '<', valueFilter : 'date',
+    },
+    {
+      key : 'entry_date_from', label : 'STOCK.ENTRY_DATE', comparitor : '>', valueFilter : 'date',
+    },
+    {
+      key : 'entry_date_to', label : 'STOCK.ENTRY_DATE', comparitor : '<', valueFilter : 'date',
+    },
+    {
+      key : 'expiration_date_from', label : 'STOCK.EXPIRATION_DATE', comparitor : '>', valueFilter : 'date',
+    },
+    {
+      key : 'expiration_date_to', label : 'STOCK.EXPIRATION_DATE', comparitor : '<', valueFilter : 'date',
+    },
+  ];
+
+  class StockFilterer {
+    constructor(cacheKey = 'stock-filterer-cache') {
```

Comment:
Since this Class is for Stock, if we omit to specify the `cacheKey` the class will just use this default `cacheKey` of stock for its appcache instance.

Another thing is we must be able to use different caches for stock registries.